flag.go

```diff
@@ -174,18 +176,34 @@ func sortFlags(flags map[normalizedName]*Flag) []*Flag {
 	list.Sort()
 	result := make([]*Flag, len(list))
 	for i, name := range list {
-		result[i] = flags[normalizedName(name)]
+		result[i] = flags[NormalizedName(name)]
 	}
 	return result
 }
 
-func (f *FlagSet) normalizeFlagName(name string) normalizedName {
+func NormalizeWordSeparators(f *FlagSet, name string) NormalizedName {
 	result := name
 	for _, sep := range f.wordSeparators {
 		result = strings.Replace(result, sep, "-", -1)
 	}
 	// Type convert to indicate normalization has been done.
-	return normalizedName(result)
+	return NormalizedName(result)
+}
+
+func (f *FlagSet) SetNormalizeName(n func(f *FlagSet, name string) NormalizedName) {
+	f.normalizeNameFunc = n
+}
+
+func (f *FlagSet) GetNormalizeName() func(f *FlagSet, name string) NormalizedName {
+	if f.normalizeNameFunc != nil {
+		return f.normalizeNameFunc
+	}
+	return NormalizeWordSeparators
```

Comment:
If we do this, and I was leaning towards it when my PR got merged, we should just do it, and not carry the WordSeparators changes.  It's too complex for too little value to have both.